### PR TITLE
[feature](function) support ip function ipv6numtostring(alias inet6_ntoa)

### PR DESCRIPTION
**Pull request by superdiaodiao**
## Proposed changes

Issue Number: #21489

Issue Number: #19624

Doris output:

image

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Review by morningman — DISMISSED** [review]

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

File: be/src/vec/functions/function_ip.h [hunk 1]
```diff
@@ -250,4 +252,98 @@ class FunctionIPv4StringToNum : public IFunction {
...
+        std::vector<IPv6Address> ipv6_addresses(input_rows_count);
+
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            bool is_empty = false;
+
+            if (col_ipv6) {
+                const auto& vec_in = col_ipv6->get_data();
+                memcpy(ipv6_addresses[i].data, reinterpret_cast<const unsigned char*>(&vec_in[i]),
+                       IPV6_BINARY_LENGTH);
+            } else {
```
Comment (HappenLee) [inline be/src/vec/functions/function_ip.h:312]: which case not `col_ipv6` ？

File: be/src/vec/functions/function_ip.h [hunk 2]
```diff
@@ -250,4 +252,98 @@ class FunctionIPv4StringToNum : public IFunction {
...
+
+        return make_nullable(std::make_shared<DataTypeString>());
+    }
+
+    bool use_default_implementation_for_nulls() const override { return true; }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) const override {
+        ColumnPtr column = block.get_by_position(arguments[0]).column;
+        const auto* col_ipv6 = check_and_get_column<ColumnIPv6>(column.get());
```
Comment (HappenLee) [inline be/src/vec/functions/function_ip.h:280]: better use two different function to dispose different type，more effective and readable

File: be/src/vec/functions/function_ip.h [hunk 3]
```diff
@@ -250,4 +252,98 @@ class FunctionIPv4StringToNum : public IFunction {
...
+        auto* begin = reinterpret_cast<char*>(vec_res.data());
+        auto* pos = begin;
+
+        auto null_map = ColumnUInt8::create(input_rows_count, 0);
+
+        struct IPv6Address {
+            unsigned char data[IPV6_BINARY_LENGTH];
+        };
+
+        std::vector<IPv6Address> ipv6_addresses(input_rows_count);
```
Comment (HappenLee) [inline be/src/vec/functions/function_ip.h:303]: maybe can reuse the mem，no need use a vector？ think again？

File: be/src/vec/functions/function_ip.h [hunk 4]
```diff
@@ -250,4 +252,98 @@ class FunctionIPv4StringToNum : public IFunction {
...
+                            arguments[0]->get_name(), get_name());
+
+        return make_nullable(std::make_shared<DataTypeString>());
+    }
+
+    bool use_default_implementation_for_nulls() const override { return true; }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) const override {
+        ColumnPtr column = block.get_by_position(arguments[0]).column;
```
[the remark this answers is not in the document]
Comment (superdiaodiao) [reply be/src/vec/functions/function_ip.h]: OK

File: be/src/vec/functions/function_ip.h (hunk 2 above)
Comment (superdiaodiao) [reply be/src/vec/functions/function_ip.h:331]: OK, more reasonable.

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run buildall

**Comment by superdiaodiao** [comment]
run arm

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]

**Pull request merged**